core/spot_termination.go

```diff
@@ -203,3 +204,58 @@ func (s *SpotTermination) asgHasTerminationLifecycleHook(autoScalingGroupName *s
 
 	return hasHook
 }
+
+// Checks to see whether an instance is in an AutoSpotting ASG as defined by tags
+// If the ASG does not have the required tags, it is not an AutoSpotting ASG and should be left
+func (s *SpotTermination) isInAutoSpottingASG(instanceID *string, tagFilteringMode string, filterByTags string) bool {
+	var optInFilterMode = (tagFilteringMode != "opt-out")
+
+	asgName, err := s.getAsgName(instanceID)
+
+	if err != nil {
+		logger.Printf("Failed get ASG name for %s with err: %s\n", *instanceID, err.Error())
+		return false
+	}
+
+	logger.Printf("starting logic for ASG")
+	asgGroupsOutput, err := s.asSvc.DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: []*string{
+			&asgName,
+		},
+	})
+
+	if err != nil {
+		logger.Printf("Failed to get ASG using ASG name %s with err: %s\n", asgName, err.Error())
+		return false
+	}
+
+	filters := replaceWhitespace(filterByTags)
+
+	var tagsToMatch = []Tag{}
+
+	if len(filters) == 0 {
+		tagsToMatch = []Tag{{Key: "spot-enabled", Value: "true"}}
+	}
+
+	for _, tagWithValue := range strings.Split(filters, ",") {
+		tag := splitTagAndValue(tagWithValue)
+		if tag != nil {
+			tagsToMatch = append(tagsToMatch, *tag)
+		}
+	}
+
+	if len(tagsToMatch) == 0 {
+		tagsToMatch = []Tag{{Key: "spot-enabled", Value: "true"}}
```

Comment:
I think this should not be hard-coded to the default tag but made to obey the tag filter configuration that can have different tags.

Reply:
Have removed the hard code checks